qendpoint-store/src/test/java/com/the_qa_company/qendpoint/store/EndpointSPARQL11QueryComplianceTest.java

```diff
@@ -1,124 +1,127 @@
+//import org.eclipse.rdf4j.repository.sail.SailRepository;
+//import org.eclipse.rdf4j.testsuite.query.parser.sparql.manifest.SPARQL11QueryComplianceTest;
+//import org.junit.Rule;
+//import org.junit.rules.TemporaryFolder;
+//import com.the_qa_company.qendpoint.core.enums.RDFNotation;
+//import com.the_qa_company.qendpoint.core.exceptions.NotFoundException;
+//import com.the_qa_company.qendpoint.core.exceptions.ParserException;
+//import com.the_qa_company.qendpoint.core.hdt.HDT;
+//import com.the_qa_company.qendpoint.core.hdt.HDTManager;
+//import com.the_qa_company.qendpoint.core.options.HDTOptions;
+//import com.the_qa_company.qendpoint.core.options.HDTOptionsKeys;
+//import org.slf4j.Logger;
+//import org.slf4j.LoggerFactory;
+//
+//import java.io.File;
+//import java.io.IOException;
+//import java.net.JarURLConnection;
+//import java.net.URL;
+//import java.util.ArrayList;
+//import java.util.List;
+//
+///**
+// * @author Ali Haidar
+// */
+//public class EndpointSPARQL11QueryComplianceTest extends SPARQL11QueryComplianceTest {
+//	private static final Logger logger = LoggerFactory.getLogger(EndpointSPARQL11QueryComplianceTest.class);
+//
+//	public EndpointSPARQL11QueryComplianceTest(String displayName, String testURI, String name, String queryFileURL,
+//			String resultFileURL, Dataset dataset, boolean ordered, boolean laxCardinality)
+//			throws ParserException, NotFoundException, IOException {
+//		super(displayName, testURI, name, queryFileURL, resultFileURL, null, ordered, laxCardinality);
+//		setUpHDT(dataset);
+//		List<String> testToIgnore = new ArrayList<>();
+//		// @todo these tests are failing and should not, they are skipped so
+//		// that we can be sure that we see when
+//		// currently passing tests are not failing. Many of these tests are not
+//		// so problematic since we do not support
+//		// named graphs anyway
+//		testToIgnore.add("constructwhere02 - CONSTRUCT WHERE");
+//		testToIgnore.add("constructwhere03 - CONSTRUCT WHERE");
+//		testToIgnore.add("constructwhere04 - CONSTRUCT WHERE");
+//		testToIgnore.add("Exists within graph pattern");
+//		testToIgnore.add("(pp07) Path with one graph");
+//		testToIgnore.add("(pp35) Named Graph 2");
+//		testToIgnore.add("sq01 - Subquery within graph pattern");
+//		testToIgnore.add("sq02 - Subquery within graph pattern, graph variable is bound");
+//		testToIgnore.add("sq03 - Subquery within graph pattern, graph variable is not bound");
+//		testToIgnore.add("sq04 - Subquery within graph pattern, default graph does not apply");
+//		testToIgnore.add("sq05 - Subquery within graph pattern, from named applies");
+//		testToIgnore.add("sq06 - Subquery with graph pattern, from named applies");
+//		testToIgnore.add("sq07 - Subquery with from ");
+//		testToIgnore.add("sq11 - Subquery limit per resource");
+//		testToIgnore.add("sq13 - Subqueries don't inject bindings");
+//		testToIgnore.add("sq14 - limit by resource");
+//
+//		this.setIgnoredTests(testToIgnore);
+//	}
```

Comment:
The test class in RDF4J has changed quite a bit in RDF4J 5.0.0 and there is no longer a constructor that provides the dataset. I've simply commented out the code since I wasn't sure how to fix it in the RDF4J code.

Reply:
It would be good to understand how it works before merging

Reply:
I took some time to figure out the RDF4J side of things and I've created a method that gets called with all the test arguments so that we can listen in and read the dataset variable to set up the HDT store.